src/main/java/graphql/execution/reactive/CompletionStageMappingPublisher.java

```diff
@@ -52,126 +54,4 @@ public Publisher<U> getUpstreamPublisher() {
-        final AtomicReference<Runnable> onCompleteOrErrorRun;
-        final AtomicBoolean onCompleteOrErrorRunCalled;
-
-        public CompletionStageSubscriber(Subscriber<? super D> downstreamSubscriber) {
-            this.downstreamSubscriber = downstreamSubscriber;
-            inFlightDataQ = new ArrayDeque<>();
-            onCompleteOrErrorRun = new AtomicReference<>();
-            onCompleteOrErrorRunCalled = new AtomicBoolean(false);
-        }
-
-
-        @Override
-        public void onSubscribe(Subscription subscription) {
-            delegatingSubscription = new DelegatingSubscription(subscription);
-            downstreamSubscriber.onSubscribe(delegatingSubscription);
-        }
-
-        @Override
-        public void onNext(U u) {
-            // for safety - no more data after we have called done/error - we should not get this BUT belts and braces
-            if (onCompleteOrErrorRunCalled.get()) {
-                return;
-            }
-            try {
-                CompletionStage<D> completionStage = mapper.apply(u);
-                offerToInFlightQ(completionStage);
-                completionStage.whenComplete(whenNextFinished(completionStage));
-            } catch (RuntimeException throwable) {
-                handleThrowable(throwable);
-            }
-        }
-
-        private BiConsumer<D, Throwable> whenNextFinished(CompletionStage<D> completionStage) {
-            return (d, throwable) -> {
-                try {
-                    if (throwable != null) {
-                        handleThrowable(throwable);
-                    } else {
-                        downstreamSubscriber.onNext(d);
-                    }
-                } finally {
-                    Runnable runOnCompleteOrErrorRun = onCompleteOrErrorRun.get();
-                    boolean empty = removeFromInFlightQAndCheckIfEmpty(completionStage);
-                    if (empty && runOnCompleteOrErrorRun != null) {
-                        onCompleteOrErrorRun.set(null);
-                        runOnCompleteOrErrorRun.run();
-                    }
-                }
-            };
-        }
-
-        private void handleThrowable(Throwable throwable) {
-            downstreamSubscriber.onError(throwable);
-            //
-            // reactive semantics say that IF an exception happens on a publisher
-            // then onError is called and no more messages flow.  But since the exception happened
-            // during the mapping, the upstream publisher does not no about this.
-            // so we cancel to bring the semantics back together, that is as soon as an exception
-            // has happened, no more messages flow
-            //
-            delegatingSubscription.cancel();
-        }
-
-        @Override
-        public void onError(Throwable t) {
-            onCompleteOrError(() -> {
-                onCompleteOrErrorRunCalled.set(true);
-                downstreamSubscriber.onError(t);
-            });
```

Comment:
See in the old impl we waited until ALL inflights finished before we called onError - well now we dont.

We fail fast after discussing it with the Spring folks